### DISCUSSION TITLE
能给个纯 schema 模式通过配置 x-reactions 修改 x-index 的示例或语句吗

### DISCUSSION THREAD

**Discussion by prettyboyweiwei** [alibaba/formily#3507] [Q&A 2.x]
已经尝试了 schema 配置和代码里 take().setState 均无效，其他属性倒是可以，唯独 x-index 不行。我看 schema 里的 x-index 和 Field 的 index 也不是同一个意思。求示例
```json
{
  "x-index": 1,
  "x-reactions": {
    "dependencies": [
      {
        "property": "value",
        "type": "any",
        "name": "value"
      }
    ],
    "fulfill": {
      "state": {
        "index": "{{$deps.value}}"
      },
      "schema": {
        "x-index": "{{$deps.value}}"
      }
    }
  }
}
```
```javascript
form
  .query('input1')
  .take()
  .setState((state) => {
    state.index = state.value;
    state['x-index'] = state.value;
  });
```

**Comment by frehaiku** [comment]
x-index 没有对应的 Field 属性，待支持；可以转成 issue

**Reply by janryWang** [reply]
这个相当于改schema结构，需要走受控表单的方式来更新schema

**Reply by frehaiku** [reply]
我们也有类似的需求，同层级兄弟元素动态排序，场景是：
select multiple 多选时，根据value选中元素的顺序对兄弟元素进行排序

select选中了[c,b,a]，c/b/a表示 field path
下方的input框，按照c、b、a排序

**Reply by prettyboyweiwei** [reply]
在表单较多，千人千面定制自己哪些是常用字段时，排序和隐藏是很常见的需求。
目前对于排序我是使用改变 schema json 的方式实现的。期望后续能完全依靠 schema 本身实现。
因为我的理解是 schema === 开发写的表单代码，代码始终是一份，不会根据不同用户的展现不同而有任何区别，而是表单通过解析另外一份配置数据来初始化这些不同。那么 schema 本身也应该能包含表单如何变化和联动逻辑，做到所有人都是同一份 schema，那份数据会作为 schema Form 的某些初始状态。
